sympy/solvers/solveset.py

```diff
@@ -958,6 +958,11 @@ def solveset(f, symbol=None, domain=S.Complexes):
                 not supported. Try the real domain by
                 setting domain=S.Reals'''))
         try:
+            for e in f.args:
+                    n, d = ratsimp(e).as_numer_denom()
+                    if sympify(d) == symbol:
```

Comment:
The results of sympy functions are already sympified so you can just write `if d == symbol:`.